tests/test_fabricpath_global.rb

```diff
@@ -42,213 +40,203 @@ def no_feature_fabricpath
-    assert_equal(:disable, @global.graceful_merge,
-                 'graceful merge not set to disable')
+    fg = FabricpathGlobal.new('default')
+    assert_equal(fg.default_graceful_merge, fg.graceful_merge)
+
+    fg.graceful_merge = :disable
+    assert_equal(:disable, fg.graceful_merge)
   end
 
   def test_linkup_delay_all
-    @global = FabricpathGlobal.new('default')
-    # test default value
-    assert_equal(@global.default_linkup_delay,
-                 @global.linkup_delay,
-                 'Default linkup_delay not set correctly')
-    @global.linkup_delay = 25
-    assert_equal(25, @global.linkup_delay,
-                 'linkup_delay not set to 25')
-
-    return if n5k6k_platforms?
-
-    refute(@global.linkup_delay_always,
-           'linkup_delay_always should not be set by default')
-    @global.linkup_delay_always = true
-    assert(@global.linkup_delay_always,
-           'linkup_delay_always is not getting set')
-
-    @global.linkup_delay_enable = true
-    @global.linkup_delay_enable = false
-    refute(@global.linkup_delay_enable,
-           'linkup_delay is not getting disabled')
+    fg = FabricpathGlobal.new('default')
+    assert_equal(fg.default_linkup_delay, fg.linkup_delay)
+
+    fg.linkup_delay = 25
+    assert_equal(25, fg.linkup_delay)
+
+    if validate_property_excluded?('fabricpath', 'linkup_delay_always')
+      assert_raises(Cisco::UnsupportedError) { fg.linkup_delay_always = true }
+    else
+      assert_equal(fg.default_linkup_delay_always, fg.linkup_delay_always)
+      fg.linkup_delay_always = true
+      assert(fg.linkup_delay_always, 'linkup_delay_always: Expected: true')
+      fg.linkup_delay_always = false
+      refute(fg.linkup_delay_always, 'linkup_delay_always: Expected: false')
+    end
+
+    if validate_property_excluded?('fabricpath', 'linkup_delay_enable')
+      assert_raises(Cisco::UnsupportedError) { fg.linkup_delay_enable = true }
+      return
+    end
+
+    assert_equal(fg.default_linkup_delay_enable, fg.linkup_delay_enable)
+    fg.linkup_delay_enable = true
+    assert(fg.linkup_delay_enable, 'linkup_delay_enable: Expected: true')
+    fg.linkup_delay_enable = false
+    refute(fg.linkup_delay_enable, 'linkup_delay_enable: Expected: false')
   end
 
   def test_loadbalance_algorithm
-    @global = FabricpathGlobal.new('default')
-    # test default value
-    check_val = n5k6k_platforms? ? 'source-destination' : 'symmetric'
-    assert_equal(check_val, @global.loadbalance_algorithm,
-                 "default algo should be #{check_val} but is
-                  #{@global.loadbalance_algorithm}")
-    @global.loadbalance_algorithm = 'source'
-    assert_equal('source', @global.loadbalance_algorithm,
-                 'algo not getting set to source-destination')
+    fg = FabricpathGlobal.new('default')
+    default =
+      cmd_ref.lookup('fabricpath', 'loadbalance_algorithm').default_value
+    assert_equal(default, fg.loadbalance_algorithm)
```

Comment:
Good question. I think that may have been an earlier workaround that I should've removed. I'll fix it.